src/midi/midi_lasynth_model.cpp

```diff
@@ -50,48 +51,72 @@ LASynthModel::LASynthModel(const std::string &rom_name,
 	assert(ctrl_full || (ctrl_a && ctrl_b));
 }
 
+// Scans the requested directory for the requested ROM.
+// Uses a static cache of directories and their ROMs to avoid repeat scans.
 std::optional<std_fs::path> LASynthModel::find_rom(const Mt32ServicePtr& service,
                                                    const std_fs::path& dir,
                                                    const Rom* rom)
 {
-	static std::unordered_set<std::string> unknown_files;
 	if (!rom) {
-		return {};
+		return std::nullopt;
 	}
 
-	std::error_code ec;
+	// Named types to make the directory cache self-documenting
+	using RomId     = std::string;
+	using RomPath   = std_fs::path;
+	using DirPath   = std_fs::path;
+	using RomsCache = std::map<RomId, RomPath>;
+	using DirsCache = std::map<DirPath, RomsCache>;
+	using JunkFiles = std::set<RomPath>;
```

Comment:
IMO, the path aliases just obscure the intent, so I'd suggest getting rid of them.

I also have a slight preference for `UnknownFiles` vs `JunkFiles`, but we're splitting hairs here 😆 

```cpp
using RomId        = std::string;
using RomsCache    = std::map<RomId, std_fs::path>;
using DirsCache    = std::map<std_fs::path, RomsCache>;
using UnknownFiles = std::set<std_fs::path>;
```